src/main/java/com/hillel/items_exchange/controller/LocationController.java

```diff
@@ -0,0 +1,87 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.LocationDto;
+import com.hillel.items_exchange.mapper.UtilMapper;
+import com.hillel.items_exchange.service.LocationService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.boot.model.naming.IllegalIdentifierException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.constraints.PositiveOrZero;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.hillel.items_exchange.config.SecurityConfig.HAS_ROLE_ADMIN;
+import static com.hillel.items_exchange.util.MessageSourceUtil.getExceptionMessageSourceWithAdditionalInfo;
+import static com.hillel.items_exchange.util.MessageSourceUtil.getExceptionMessageSourceWithId;
+
+@RestController
+@RequestMapping("/location")
+@RequiredArgsConstructor
+@Validated
+@Slf4j
+public class LocationController {
+    private final LocationService locationService;
+
+    @GetMapping
+    @ApiOperation(value = "Get all of existed locations.")
+    public ResponseEntity<List<LocationDto>> getAllLocations() {
+        return new ResponseEntity<>(locationService.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("/{location_id}")
+    @ApiOperation(value = "Get an existed location by its ID.")
+    public ResponseEntity<LocationDto> getLocation(@PathVariable("location_id")
+                                                   @PositiveOrZero(message = "{invalid.id}") long id) {
+        return ResponseEntity.of(locationService.getById(id));
+    }
+
+    @PreAuthorize(HAS_ROLE_ADMIN)
+    @PostMapping
+    @ApiOperation(value = "Save a new Location. Only for users authorized as admin.")
+    public ResponseEntity<LocationDto> createLocation(@Valid @RequestBody LocationDto locationDto) {
+        long id = locationDto.getId();
+        if (id != 0) {
+            String message = String.format("Location not created. Received ID=%s", id);
+            log.warn(message);
```

Comment:
You could make it in such way: `log.warn("Location not created. Received ID={}", id);` and logger will add ID value into.